W-12521538-changeReqToRespHTTPcomponent-duke (#2460)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/about-mule-message.adoc

The Mule message is immutable, so every change to a Mule message results in the creation of a new instance. Each processor in a flow that receives a message returns a new Mule message consisting of these parts:

* A message `payload`, which contains the body of the message. For example: the content of a file, the HTTP body, a record from a database, or the response to a REST or Web Service request.
* Message `attributes`, which are metadata associated with the payload, such as HTTP headers and query parameters.

image::mule-message-structure-82af1.png[]

> A Mule event source triggers a process that generates Mule event with a Mule message and executes the Mule flow that contains the event source. For example, this process begins each time an xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 

Author: ```suggestion
A Mule event source triggers a process that creates a Mule event with a Mule message and executes the Mule flow that contains the event source. For example, this process begins each time an xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 
```

> A Mule event source triggers a process that generates Mule message, incorporates the message into a Mule event, and executes the Mule flow that contains the event source. For example, this process begins each time an xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 

Author: ```suggestion
A Mule event source triggers a process that generates Mule event with a Mule message and executes the Mule flow that contains the event source. For example, this process begins each time an xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 
```

> A Mule event source triggers a process that generates Mule message, incorporates the message into a Mule event, and executes the Mule flow that contains the event source. For example, this process begins each time an  xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 

Author: ```suggestion
A Mule event source triggers a process that generates Mule message, incorporates the message into a Mule event, and executes the Mule flow that contains the event source. For example, this process begins each time an xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 
```

> Triggering a Mule event source initiates a process that generates a Mule message, places the message in a Mule event, and executes the Mule flow that contains the event source. For example, this process begins each time an  xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 

Author: ```suggestion
Execution of a Mule event source initiates a process that generates a Mule message, places the message in a Mule event, and starts the Mule flow that contains the event source. For example, this process begins each time an  xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 
```

> Execution of a Mule event source initiates a process that generates a Mule message, places the message in a Mule event, and starts the Mule flow that contains the event source. For example, this process begins each time an  xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 

Author: ```suggestion
Execution of a Mule event source initiates a process that triggers the creation of a Mule message, places the message in a Mule event, and starts the Mule flow that contains the event source. For example, this process begins each time an  xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 
```

> Execution of a Mule event source initiates a process that triggers the creation of a Mule message, places the message in a Mule event, and starts the Mule flow that contains the event source. For example, this process begins each time an  xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 

Reviewer: I would remove "Execution of" and I think all the clarifications about the mule flow "that contains the event source" makes it more difficult to understand and has no use for the end user. The flow doesn't start in that moment but is executed. I would simply  add this:
"A mule event source triggers a process which involves the creation of an event and the execution of the flow associated to it. For example, this process begins.."

Author: @fsgonz This page relates to the Mule message, so the goal is to say something about the relationship of the message to the execution of the event source if at all possible. This is to help clarify language elsewhere that the "event source triggers a Mule message". 

Is there a way to say something about the Mule message in this process?

Reviewer: If you define that an event has a message, it implies from the event source, the created event will have a message. I think there is nothing strictly incorrect about the paragraph so it's ok for me if it seems clearer for the users.

Author: ```suggestion
A Mule event source triggers a process that generates Mule message, incorporates the message into a Mule event, and executes the Mule flow that contains the event source. For example, this process begins each time an  xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source receives a request or the xref:scheduler-concept.adoc[Scheduler] executes. 
```

> Once the flow begins its execution, downstream processors (such as Core components, File read operations, or the HTTP request operation) can then retrieve, set, and process Mule message data (payload and attributes) that reside in the Mule event according to their configurations.

Reviewer 2: ```suggestion
After the flow begins its execution, downstream processors (such as Core components, File read operations, or the HTTP request operation) can then retrieve, set, and process Mule message data (payload and attributes) that reside in the Mule event according to their configurations.
```

> Once the flow starts, downstream processors (such as Core components, File read operations, or the HTTP request operation) can then retrieve, set, and process Mule message data (payload and attributes) that reside in the Mule event according to their configurations.

Reviewer: Once the flow starts -> Once the flow begins its execution

> Once the flow starts, downstream processors (such as Core components, File read operations, or the HTTP request operation) can then retrieve, set, and process Mule message data (payload and attributes) that reside in the Mule event according to their configurations.

Author: ```suggestion
Once the flow begins its execution, downstream processors (such as Core components, File read operations, or the HTTP request operation) can then retrieve, set, and process Mule message data (payload and attributes) that reside in the Mule event according to their configurations.
```

image::mule-message-structure-82af1.png[]

A Mule message is created as part of a xref:about-mule-event.adoc[Mule event] during the process in which a Mule event source in a Mule flow triggers a flow to start, as when an HTTP Listener component receives a request or each time the Scheduler component triggers an execution of the flow.

> For example, a request to an xref:xref:connectors::http/http-listener-ref.adoc[HTTP Listener] component in a Mule app triggers a Mule message. Message processors in the flow (such as Core components, File read operations, or the HTTP request operations) can then retrieve, set, and process Mule message data (payload and attributes) that reside in the Mule event according to their configurations.

Author: ```suggestion
For example, a request to an xref:xref:connectors::http/http-listener-ref.adoc[HTTP Listener] source in a Mule application triggers a Mule message. Message processors in the flow (such as Core components, File read operations, or the HTTP request operation) can then retrieve, set, and process Mule message data (payload and attributes) that reside in the Mule event according to their configurations.
```

> For example, a request to an HTTP Listener component in a Mule app triggers a Mule message that contains the body of the HTTP response as its payload, along with the metadata associated with that content, such as HTTP headers, as it attributes. Message processors in the flow (such as Core components, File read operations, or the HTTP request operations) can then retrieve, set, and process Mule message data that resides in the Mule event according to their configurations.

Author: ```suggestion
For example, a request to an xref:xref:connectors::http/http-listener-ref.adoc[HTTP Listener] component in a Mule app triggers a Mule message. Message processors in the flow (such as Core components, File read operations, or the HTTP request operations) can then retrieve, set, and process Mule message data (payload and attributes) that reside in the Mule event according to their configurations.
```
